### ISSUE TITLE
Playbook execution does not say failed even when there is authentication error.

### ISSUE THREAD

**Issue by AseemKumar** [ansible/ansible#16877]
##### ISSUE TYPE

- Bug Report
##### ANSIBLE VERSION

ansible 2.1.0.0
##### CONFIGURATION

default config
##### OS / ENVIRONMENT

ansible is running on Ubuntu Trusty and the systems being managed are Oracle linux.
##### SUMMARY

Fetching a file from many servers and if the password is incorrect, then I do not get failed=1 for those servers. For servers that are authenticated and file is fetched, I get ServerName: ok=1    changed=1    unreachable=0    failed=0.
And for those to which I supply wrong password, I get ServerName: ok=1    changed=0    unreachable=0    failed=0
##### STEPS TO REPRODUCE

Executing the below given playbook

```
- hosts: '{{ target }}'
  gather_facts: no
  tasks:
  - name: Copy passwd file to ansible server.
    fetch: src=/etc/passwd dest=/home/vagrant/gatheredData/password_file
```
##### EXPECTED RESULTS

I think it should give failed=1 as the command could not be executed against that server.
Also, why does it give changed=1 for fetching file, as nothing was changed on the server?
##### ACTUAL RESULTS

I got only ok=1 and with verbose I get the message "unable to calculate the checksum of the remote file"

```
Using /etc/ansible/ansible.cfg as config file
Loaded callback default of type stdout, v2.0

PLAYBOOK: GetPasswordFile.yml **************************************************
1 plays in GetPasswordFile.yml

PLAY [ServerName] ************************************

TASK [Copy passwd file to ansible server.] *************************************
task path: /home/vagrant/ansible/playbooks/GetPasswordFile.yml:7
<ServerName> ESTABLISH SSH CONNECTION FOR USER: UserName
<ServerName> SSH: ansible.cfg set ssh_args: (-o)(ControlMaster=auto)(-o)(ControlPersist=60s)
<ServerName> SSH: ANSIBLE_HOST_KEY_CHECKING/host_key_checking disabled: (-o)(StrictHostKeyChecking=no)
<ServerName> SSH: ANSIBLE_REMOTE_PORT/remote_port/ansible_port set: (-o)(Port=22)
<ServerName> SSH: ANSIBLE_REMOTE_USER/remote_user/ansible_user/user/-u set: (-o)(User=UserName)
<ServerName> SSH: ANSIBLE_TIMEOUT/timeout set: (-o)(ConnectTimeout=30)
<ServerName> SSH: PlayContext set ssh_common_args: ()
<ServerName> SSH: PlayContext set ssh_extra_args: ()
<ServerName> SSH: found only ControlPersist; added ControlPath: (-o)(ControlPath=/home/vagrant/.ansible/cp/ansible-ssh-%h-%p-%r)
<ServerName> SSH: EXEC sshpass -d12 ssh -C -vvv -o ControlMaster=auto -o ControlPersist=60s -o StrictHostKeyChecking=no -o Port=22 -o User=UserName -o ConnectTimeout=30 -o ControlPath=/home/vagrant/.ansible/cp/ansible-ssh-%h-%p-%r ServerName '/bin/sh -c '"'"'( umask 77 && mkdir -p "` echo $HOME/.ansible/tmp/ansible-tmp-1469812752.41-71145122664819 `" && echo ansible-tmp-1469812752.41-71145122664819="` echo $HOME/.ansible/tmp/ansible-tmp-1469812752.41-71145122664819 `" ) && sleep 0'"'"''
ok: [ServerName] => {"changed": false, "file": "/etc/passwd", "invocation": {"module_args": {"dest": "/home/vagrant/gatheredData/password_file", "src": "/etc/passwd"}, "module_name": "fetch"}, "msg": "unable to calculate the checksum of the remote file"}

PLAY RECAP *********************************************************************
ServerName : ok=1    changed=0    unreachable=0    failed=0

```

**Comment by alikins** [comment]
Could you test this with ansible-2.1.1 ( https://github.com/ansible/ansible/releases/tag/v2.1.1.0-1 )
There have been a few bug fixes related to this.

Changelog.md snippets...

> - Fixed several more bugs in relation to the way play execution continues or halts when hosts fail, to bringe the behavior more in line with 1.9.x.
> - Fix for bug related to unreachable hosts and any_errors_fatal in the linear strategy
> - Fixed a bug in which the number of retries in a do/until loop was off by one
> - More fixes to the way failed and unreachable hosts affect future plays

**Comment by AseemKumar** [comment]
Hi @alikins , I tested with 2.1.1 version. I put in a total of 11 machines; 6 servers that were reachable, 3 that were unreachable and 2 that had a wrong password entry.

As a result, I got "ok=1" for 6 and "unreachable=1" for the rest 5. Shouldn't those that had wrong password show as failed instead as the message reads "Authentication failure."?

**Comment by alikins** [comment]
> As a result, I got "ok=1" for 6 and "unreachable=1" for the rest 5. Shouldn't those that had wrong password show as failed instead as the message reads "Authentication failure."?

Sounds like 2.1.1 is still busted.

**Comment by jimi-c** [comment]
@AseemKumar / @alikins bad passwords have traditionally been in the "unreachable" category, so it sounds like 2.1.1 is actually doing the correct thing. Whether those should be considered unreachable or failed is up for debate, but like I said this appears to be resolved in 2.1.1.

**Comment by jimi-c** [comment]
So based on the above, with no follow-up, I'm going to go ahead and close this. If you have any further questions, please let us know by stopping by one of the two mailing lists, as appropriate:

   * https://groups.google.com/forum/#!forum/ansible-project - for user questions, tips, and tricks
   * https://groups.google.com/forum/#!forum/ansible-devel - for strategy, future planning, and questions about writing code

Because this project is very active, we're unlikely to see comments made on closed tickets, but the mailing list is a great way to ask questions, or post if you don't think this particular issue is resolved.

Thank you!
